diff -ruNp 1.16.1-r0/nchan-1.2.8.orig/src/subscribers/websocket.c 1.16.1-r0/nchan-1.2.8/src/subscribers/websocket.c
--- 1.16.1-r0/nchan-1.2.8.orig/src/subscribers/websocket.c	2021-07-27 17:05:53.136295568 +0200
+++ 1.16.1-r0/nchan-1.2.8/src/subscribers/websocket.c	2021-07-27 17:07:34.738892359 +0200
@@ -513,7 +513,9 @@ static ngx_int_t websocket_publish_callb
 
 static void websocket_publish_continue(ws_publish_data_t *d) {
   nchan_msg_t             *msg = &d->msg;
+#if 0
   struct timeval           tv;
+#endif
   full_subscriber_t       *fsub = d->fsub;
   ngx_http_request_t      *r = d->fsub->sub.request;
   
@@ -523,9 +525,13 @@ static void websocket_publish_continue(w
   if(r->headers_in.content_type) {
     msg->content_type = &r->headers_in.content_type->value;
   }
-  
+ 
+#if 0 
   ngx_gettimeofday(&tv);
   msg->id.time = tv.tv_sec;
+#else
+  msg->id.time = 0;
+#endif
   msg->id.tag.fixed[0]=0;
   msg->id.tagcount=1;
   msg->id.tagactive=0;
